packages/connector/src/services/core/services/rates/service.test.ts

```diff
@@ -0,0 +1,165 @@
+import Koa from 'koa'
+import { createRatesService, RatesService, ConvertError } from './service'
+import { RafikiServicesFactory } from '../../factories'
+
+jest.useFakeTimers('modern')
+describe('Rates service', function () {
+  let service: RatesService
+  let requestCount = 0
+
+  const pricesLifetime = 100
+  const koa = new Koa()
+  koa.use(function (ctx) {
+    requestCount++
+    ctx.body = {
+      USD: 1.0, // base
+      XRP: 0.5,
+      NEGATIVE: -0.5,
+      ZERO: 0.0,
+      STRING: '123'
+    }
+  })
+  const server = koa.listen(3210)
+
+  beforeEach(() => {
+    jest.setSystemTime(1600000000000)
+    service = createRatesService({
+      pricesUrl: 'http://127.0.0.1:3210/',
+      pricesLifetime,
+      logger: RafikiServicesFactory.build().logger
+    })
+    requestCount = 0
+  })
+
+  afterAll(() => {
+    server.close()
+  })
+
+  describe('convert', function () {
+    it('returns the source amount when assets are alike', async () => {
+      await expect(
+        service.convert({
+          sourceAmount: 1234n,
+          sourceAsset: { code: 'USD', scale: 9 },
+          destinationAsset: { code: 'USD', scale: 9 }
+        })
+      ).resolves.toBe(1234n)
+      expect(requestCount).toBe(0)
+    })
+
+    it('scales the source amount when currencies are alike but scales are different', async () => {
+      await expect(
+        service.convert({
+          sourceAmount: 123n,
+          sourceAsset: { code: 'USD', scale: 9 },
+          destinationAsset: { code: 'USD', scale: 12 }
+        })
+      ).resolves.toBe(123_000n)
+      await expect(
+        service.convert({
+          sourceAmount: 123456n,
+          sourceAsset: { code: 'USD', scale: 12 },
+          destinationAsset: { code: 'USD', scale: 9 }
+        })
+      ).resolves.toBe(123n)
+      expect(requestCount).toBe(1)
```

Comment:
What do you think about skipping the `sharedLoad` request for same currency, different scales?